nx/lib/nx.ex

```diff
@@ -7390,13 +7430,165 @@ defmodule Nx do
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 1],
+          [0, 1, 0]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :y)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 2],
+          [0, 2, 1]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :y, comparator: :asc)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [2, 0, 1],
+          [1, 2, 0]
+        ]
+      >
+
+      iex> tensor = Nx.tensor([[[4, 5], [2, 5], [5, 0]], [[1, 9], [2, 1], [2, 1]], [[0, -1], [-1, 0], [0, -1]], [[-1, 0], [0, -1], [-1, 0]]], names: [:x, :y, :z])
+      iex> Nx.argsort(tensor, axis: :x)
+      #Nx.Tensor<
+        s64[x: 4][y: 3][z: 2]
+        [
+          [
+            [3, 2],
+            [2, 3],
+            [3, 2]
+          ],
+          [
+            [2, 3],
+            [3, 2],
+            [2, 0]
+          ],
+          [
+            [1, 0],
+            [0, 1],
+            [1, 3]
+          ],
+          [
+            [0, 1],
+            [1, 0],
+            [0, 1]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[[4, 5, 2], [2, 5, 3], [5, 0, 2]], [[1, 9, 8], [2, 1, 3], [2, 1, 4]]], names: [:x, :y, :z]), axis: :x)
+      #Nx.Tensor<
+        s64[x: 2][y: 3][z: 3]
+        [
+          [
+            [1, 0, 0],
+            [0, 1, 0],
+            [1, 0, 0]
+          ],
+          [
+            [0, 1, 1],
+            [1, 0, 1],
+            [0, 1, 1]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[[4, 5, 2], [2, 5, 3], [5, 0, 2]], [[1, 9, 8], [2, 1, 3], [2, 1, 4]]], names: [:x, :y, :z]), axis: :y)
+      #Nx.Tensor<
+        s64[x: 2][y: 3][z: 3]
+        [
+          [
+            [1, 2, 0],
+            [0, 0, 2],
+            [2, 1, 1]
+          ],
+          [
+            [0, 1, 1],
+            [1, 2, 2],
+            [2, 0, 0]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[[4, 5, 2], [2, 5, 3], [5, 0, 2]], [[1, 9, 8], [2, 1, 3], [2, 1, 4]]], names: [:x, :y, :z]), axis: :z)
+      #Nx.Tensor<
+        s64[x: 2][y: 3][z: 3]
+        [
+          [
+            [2, 0, 1],
+            [0, 2, 1],
+            [1, 2, 0]
+          ],
+          [
+            [0, 2, 1],
+            [1, 0, 2],
+            [1, 0, 2]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :x, comparator: &Nx.less/2)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 1],
+          [0, 1, 0]
+        ]
+      >
+  """
+  @doc type: :ndim
+  def argsort(tensor, opts \\ []) do
+    opts = keyword!(opts, axis: 0, comparator: :desc)
+    comparator = opts[:comparator]
+
+    %T{shape: shape, names: names} = tensor = to_tensor(tensor)
+    axis = Nx.Shape.normalize_axis(shape, opts[:axis], names)
+
+    impl!(tensor).argsort(%{tensor | type: {:s, 64}}, tensor,
```

Comment:
I wasn't able to make argsort work with `{:u, 64}` type on EXLA, which would be ideal, so I defaulted to `{:s, 64}`

Reply:
Which error do you get? Looking at the code it feels like it should work... on the other, numpy and jax both return int64 too.

Reply:
Torchx also return int64, so let's keep this as is.

Reply:
The error was something along the lines of "The comparator function expected 0-th argument to have shape s64[], got u64[] instead"